src/debug.ts

```diff
@@ -0,0 +1,220 @@
+  if (!executable) {
+    return;
+  }
+  const { argsArray, envMap } = await getArgsAndEnv();
+
+  const newConfig = provider.createDebugConfig(name, executable, argsArray, envMap);
+  await saveConfig(newConfig);
+}
+
+async function chapel_createDebugConfigForActiveFile() {
+  const provider = getProvider();
+  if (!provider) {
+    return
+  }
+
+  // get the current file
+  const editor = vscode.window.activeTextEditor;
+  if (!editor) {
+    vscode.window.showErrorMessage("No active editor found.");
+    return;
+  }
+  const document = editor.document;
+  if (document.languageId !== "chapel") {
+    vscode.window.showErrorMessage("The current file is not a Chapel file.");
+    return;
+  }
+  const executablePath = document.fileName?.split(".").slice(0, -1).join(".");
+  if (!executablePath) {
+    vscode.window.showErrorMessage("Failed to get the file name.");
+    return;
+  }
+  const wsPath = vscode.workspace.workspaceFolders?.[0]?.uri.fsPath;
+  if (!wsPath) {
+    vscode.window.showErrorMessage("No workspace folder found.");
+  }
+  let executableNiceName: string = executablePath.split("/").pop() || executablePath;
+  let executableRelPath: string = executablePath;
+  if (wsPath) {
+    // determine executableRelPath to wsPath
+    executableNiceName = executablePath.replace(wsPath + "/", "");
+    executableRelPath = "${workspaceFolder}/" + executableNiceName;
+  }
+
+  const { argsArray, envMap } = await getArgsAndEnv();
+
+  const newConfig = provider.createDebugConfig(`Debug ${executableNiceName}`, executableRelPath, argsArray, envMap);
+  await saveConfig(newConfig);
+}
+
+
+function getProvider(): DebugProvider | undefined {
+  // determine the provider
+  const provider = getPreferredProvider();
+  if (!provider) {
+    const preferredProviderId = getPreferredDebugProvider();
+    vscode.window.showErrorMessage(`The requested debug provider '${preferredProviderId}' is not installed/supported`);
+    return;
+  }
+
+  if (!provider.isAvailable()) {
+    vscode.window.showErrorMessage(`The requested debug provider '${provider.id}' is not installed. Please install [${provider.name}](command:${provider.getInstallCommand()}) or select another provider.`);
+    return;
+  }
+  return provider;
+}
+
+async function saveConfig(newConfig: vscode.DebugConfiguration) {
+  const name = newConfig.name;
+  const launchConfigs = vscode.workspace.getConfiguration("launch");
+  const configs = launchConfigs.get<any[]>("configurations") || [];
+  const existingConfigIdx = configs.findIndex((config: any) => config.name === name);
+  if (existingConfigIdx !== -1) {
+    const choice = await vscode.window.showErrorMessage(`A debug configuration named '${name}' already exists.`, "Cancel", "Overwrite");
+    if (choice !== "Cancel") {
+      configs[existingConfigIdx] = newConfig;
+    }
+  } else {
+    configs.push(newConfig);
+  }
+  launchConfigs.update("configurations", configs).then(() => {
```

Comment:
I believe this is the only place we use `then` instead of `await` like everywhere else.
Is there a reason this is better? I'd prefer to use the same thing everywhere for consistency.

Reply:
That is intentional, if `await` is used here you cannot catch/handle the error case. But `await` is used elsewhere as its not possible for the rest of the function to procede without waiting, so you just end up with massively nested `.then` clauses.